### PR TITLE
integration: Fix test_instance_manager_cpu_reservation

### PR DESCRIPTION
**Pull request by shuo-wu**
This test is not updated after changing the design of the cpu reservation feature: Right now the deprecated setting shouldn't work. And Longhorn will error out when users try to update it.
Sorry for that. Will be careful next time.

### CODE REVIEW

**Comment by shuo-wu** [comment]
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/60/

**Comment by c3y1huang** [comment]
Ref https://github.com/longhorn/longhorn/issues/2325

File: manager/integration/tests/test_settings.py [hunk 1]
```diff
@@ -312,68 +312,42 @@ def test_instance_manager_cpu_reservation(client, core_api):  # NOQA
...
-    1. Change the deprecated setting `Guaranteed Engine CPU` to `0.1`.
-       Then wait for all the IM pods to recreated and become running.
-       --> every IM pod has guaranteed CPU set correctly to 100m
-    2. Change the new settings `Guaranteed Engine Manager CPU` and
-       `Guaranteed Replica Manager CPU` to 10 and 20, respectively.
-       --> Nothing happens.
-    3. Pick up node1, set `node.engineManagerCPURequest` and
+    1. Try to change the deprecated setting `Guaranteed Engine CPU`.
+       --> The setting update should fail.
+    2. Pick up nodeA, set `node.engineManagerCPURequest` and
```
Comment (c3y1huang) [inline manager/integration/tests/test_settings.py:317]:
```suggestion
    2. Pick up node 1, set `node.engineManagerCPURequest` and
```

File: manager/integration/tests/test_settings.py [hunk 2]
```diff
@@ -312,68 +312,42 @@ def test_instance_manager_cpu_reservation(client, core_api):  # NOQA
...
+       of these IM pods matches the above milli value.
+    3. Change the new settings `Guaranteed Engine Manager CPU` and
+       `Guaranteed Replica Manager CPU` to 10 and 20, respectively.
+       Then wait for all IM pods except for the pods on nodeA restarting.
+       --> The CPU requests of the restarted IM pods equals to
            the new setting value multiply the kube node allocatable CPU.
+    4. Set the both new settings to 0.
+       --> All IM pods except for the pod on nodeA will be restarted without
+        CPU requests.
     5. Set the fields on node1 to 0.
```
Comment (c3y1huang) [inline manager/integration/tests/test_settings.py:329]:
```suggestion
    5. Set the fields on node 1 to 0.
```

File: manager/integration/tests/test_settings.py [hunk 3]
```diff
@@ -312,68 +312,42 @@ def test_instance_manager_cpu_reservation(client, core_api):  # NOQA
...
-    4. Unset the deprecated setting.
-       Then wait for all the IM pods to recreated and become running.
-       --> The CPU requests of instance manager pods on node1 is the same
-           as the node fields.
-       --> The CPU requests of other instance manager pods equals to
+       --> The IM pods on this node will be restarted. And the CPU requests
+       of these IM pods matches the above milli value.
+    3. Change the new settings `Guaranteed Engine Manager CPU` and
+       `Guaranteed Replica Manager CPU` to 10 and 20, respectively.
+       Then wait for all IM pods except for the pods on nodeA restarting.
```
Comment (c3y1huang) [inline manager/integration/tests/test_settings.py:323]:
```suggestion
       Then wait for all IM pods except for the pods on node 1 restarting.
```

File: manager/integration/tests/test_settings.py [hunk 4]
```diff
@@ -312,68 +312,42 @@ def test_instance_manager_cpu_reservation(client, core_api):  # NOQA
...
+       --> The CPU requests of the restarted IM pods equals to
            the new setting value multiply the kube node allocatable CPU.
+    4. Set the both new settings to 0.
+       --> All IM pods except for the pod on nodeA will be restarted without
```
Comment (c3y1huang) [inline manager/integration/tests/test_settings.py:327]:
```suggestion
       --> All IM pods except for the pod on node 1 will be restarted without
```

File: manager/integration/tests/test_settings.py [hunk 5]
```diff
@@ -414,43 +405,32 @@ def test_instance_manager_cpu_reservation(client, core_api):  # NOQA
...
+
+    with pytest.raises(Exception) as e:
+        client.update(em_setting, value="41")
+    assert "should be between 0 to 40" in \
+           str(e.value)
+
+    em_setting = client.by_id_setting(SETTING_GUARANTEED_ENGINE_MANAGER_CPU)
+    with pytest.raises(Exception) as e:
+        client.update(em_setting, value="35")
+        assert "the sum should not be smaller than 0% or greater than 40%" in \
```
Comment (c3y1huang) [inline manager/integration/tests/test_settings.py:432]:
```suggestion
    assert "the sum should not be smaller than 0% or greater than 40%" in \
```

**Review by c3y1huang — CHANGES_REQUESTED** [review]
Just some wordings and indent need to resolve. Others LGTM.

**Review by innobead — DISMISSED** [review]
LGTM (btw @c3y1huang  has some comments to resolve).

I think different test cases combined in one test function are not easy for tracing although this is the current implementation already, also I know another reason is we want to leverage the function description for the test case description. 

We can revisit all test cases later to enhance a bit w/ more logging and comments for easy maintenance.

**Comment by shuo-wu** [comment]
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/73/

**Review by c3y1huang — APPROVED** [review]

**Pull request merged**